src/fsharp/vs/service.fs

```diff
@@ -504,13 +504,22 @@ type TypeCheckInfo
     let FilterItemsForCtors filterCtors items = 
         let items = items |> List.filter (function (Item.CtorGroup _) when filterCtors = ResolveTypeNamesToTypeRefs -> false | _ -> true) 
         items
-        
-    /// Looks at the exact name resolutions that occurred during type checking
-    /// If 'membersByResidue' is specified, we look for members of the item obtained 
-    /// from the name resolution and filter them by the specified residue (?)
-    let GetPreciseItemsFromNameResolution(line,colAtEndOfNames,membersByResidue,filterCtors, resolveOverloads, hasTextChangedSinceLastTypecheck) = 
-        let endOfNamesPos = Pos.fromVS line colAtEndOfNames
-        Trace.PrintLine("CompilerServicesVerbose", fun () -> sprintf "GetPreciseItemsFromNameResolution: line = %d, colAtEndOfNames = %d, endOfNamesPos = %s\n" line colAtEndOfNames (stringOfPos endOfNamesPos))
+    
+    // Filter items to show only valid & return Some if there are any
+    let ReturnItemsOfType items g denv (m:range) filterCtors hasTextChangedSinceLastTypecheck f =
+        let items = items |> RemoveDuplicateItems g
+        let items = items |> RemoveExplicitlySuppressed g
+        let items = items |> FilterItemsForCtors filterCtors
```

Comment:
use only one binding? maybe with another name instead of hiding previous `items`
```fsharp
let items = 
    items 
    |> RemoveDuplicateItems g
    |> RemoveExplicitlySuppressed g
    |> FilterItemsForCtors filterCtors